llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -6918,3 +6920,200 @@ bool CombinerHelper::matchOr(MachineInstr &MI, BuildFnTy &MatchInfo) {
 
   return false;
 }
+
+bool CombinerHelper::matchAddOverflow(MachineInstr &MI, BuildFnTy &MatchInfo) {
+  GAddCarryOut *Add = cast<GAddCarryOut>(&MI);
```

Comment:
Replacing this line with:
```
GAddSubCarryOut *Add = cast<GAddSubCarryOut>(&MI);
```
seems dangerous and defeats the purpose of the assert in the `cast` .